railties/lib/rails/generators/rails/db/system/change/change_generator.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require "rails/generators/base"
+
+module Rails
+  module Generators
+    module Db
+      module System
+        class ChangeGenerator < Base # :nodoc:
+          include Database
+          include AppName
+
+          class_option :to, required: true,
+            desc: "The DBMS to switch to."
+
+          def self.default_generator_root
+            path = File.expand_path(File.join(base_name, "app"), base_root)
+            path if File.exist?(path)
+          end
+
+          def initialize(*)
+            super
+            if !DATABASES.include?(options[:to])
+              raise Error, "Invalid value for --to option. Supported preconfigurations are: #{DATABASES.join(", ")}."
+            end
+            self.options = options.dup
+            options["database"] ||= options["to"]
+            options.freeze
```

Comment:
Perhaps we should mirror the options dup and freeze done in convert_database_option_for_jruby